### PR TITLE
Save posts, comments and likes in DB

### PR DESCRIPTION
**Pull request by elassis**
# What I have Accomplished 🏁 

### Create forms to perform the following functions:

- Creates a Post on behalf of the current_user. ✔️ 
- Create a comment on behalf of the current_user. ✔️ 
- Allow Users to add likes to Posts. ✔️

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -12,4 +12,18 @@ def show
     @post = Post.find(post_id)
     @comments = Comment.includes(:post).where("post_id = #{post_id}").references(:post)
   end
+
+  def new
+    @new_post = Post.new
+  end
+
+  def create
+    @new_post = Post.new(title: params[:title], text: params[:text], user_id: params[:user_id])
```
Comment (petudeveloper) [inline app/controllers/posts_controller.rb]: - [ ] The new post should be created using the `current_user` method, please refactor your code in order to use the requested method.
e.g. 
```ruby
@post = @current_user.posts.new(post_params) 
```
- [ ] Please remember to use [strong parameters](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides).
```ruby
    private 
    def post_params
      params.require(:post).permit(:title, :text)
    end
```

File: config/routes.rb [hunk 2]
```diff
@@ -2,7 +2,14 @@
...
   get '/users/:user_id/posts', to: 'posts#index'
-  get '/users/:user_id/posts/:id', to: 'posts#show'
+  get '/users/:user_id/posts/new', to: 'posts#new', as: 'posts_form'
+  get '/users/:user_id/posts/create', to: 'posts#create'
+  post '/users/:user_id/posts/create', to: 'posts#create', as: 'posts_create'
+  get '/users/:user_id/posts/:id/comments/create', to: 'comments#create'
+  post '/users/:user_id/posts/:id/comments/create', to: 'comments#create', as: 'comments_create'
+  get '/users/:user_id/posts/:id/likes/create', to: 'likes#create'
+  post '/users/:user_id/posts/:id/likes/create', to: 'likes#create', as: 'likes_create'
+  get '/users/:user_id/posts/:id', to: 'posts#show', as: 'post_details'
```
Comment (petudeveloper) [inline config/routes.rb]: [OPTIONAL] If you need to create routes for more than one resource, you can save a bit of typing by defining them all with a single call to `:resources`, check this [link](https://guides.rubyonrails.org/routing.html#defining-multiple-resources-at-the-same-time) of the rails guide if you are curious on how to do it.

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+class CommentsController < ActionController::Base
+  def create
+    @comment = Comment.new(text: params[:text], user_id: params[:user_id], post_id: params[:id])
```
Comment (petudeveloper) [inline app/controllers/comments_controller.rb]: - [ ] The new comment should be created using the `current_user` method, please refactor your code in order to use the requested method.
e.g. 
```ruby
    @comment = @current_user.comments.new(params)
    @comment.post_id = params[:post_id]
```
- [ ] Please remember to use [strong parameters](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides).
```ruby
    private 
    def post_params
      params.require(:post).permit(:title, :text)
    end
```

**Review by petudeveloper — CHANGES_REQUESTED** [review]
Hi @elassis, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

very close


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by AshakaE — APPROVED** [review]
### Hi @elassis   🙋‍♀️, 

# STATUS APPROVED ✔ 
   
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**